tests/test_common.py

```diff
@@ -380,3 +383,62 @@ def test_succeeded_in_one_try(self):
                 ):
                     _mapper.shutdown()
         assert all(self._succeeded_tasks)
+
+
+class Test_ensure_otaproxy_start:
+    DUMMY_SERVER_ADDR, DUMMY_SERVER_PORT = "127.0.0.1", 18888
+    DUMMY_SERVER_URL = f"http://{DUMMY_SERVER_ADDR}:{DUMMY_SERVER_PORT}"
+    LAUNCH_DELAY = 6
+
+    @staticmethod
+    def _launch_server_helper(addr: str, port: int, launch_delay: int, directory: str):
+        time.sleep(launch_delay)
+        run_http_server(addr, port, directory=directory)
+
+    @pytest.fixture
+    def subprocess_launch_server(self, tmp_path: Path):
+        (dummy_webroot := tmp_path / "webroot").mkdir(exist_ok=True)
+        _server_p = Process(
+            target=self._launch_server_helper,
+            args=[
+                self.DUMMY_SERVER_ADDR,
+                self.DUMMY_SERVER_PORT,
+                self.LAUNCH_DELAY,
+                str(dummy_webroot),
+            ],
+        )
+        try:
+            logger.info(f"wait for {self.LAUNCH_DELAY}s before launching the server")
+            _server_p.start()
+            yield
+        finally:
+            _server_p.kill()
+
+    def test_timeout_waiting(self):
+        """
+        NOTE: we intentionally not use the subprocess_launch_server fixture here
+              to let the probing timeout.
+        """
+        probing_timeout = 3
```

Comment:
to understand the relationship between LAUNCH_DELAY and probing_timeout.
```suggestion
        probing_timeout = self.LAUNCH_DELAY // 2
```

Reply:
Updated in [9da9ac1](https://github.com/tier4/ota-client/pull/219/commits/9da9ac1416d1de484a4356cbd922b4c917b32dbe)